File: packages/jx-quant/lib/common/GoldenRatioCalculator.js
var __defProp = Object.defineProperty;
var __markAsModule = (target) => __defProp(target, "__esModule", {value: true});
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, {get: all[name], enumerable: true});
};

// src/common/GoldenRatioCalculator.ts
__markAsModule(exports);
__export(exports, {
  GoldenRatioCalculator: () => GoldenRatioCalculator
});
var GoldenRatioCalculator = class {
  constructor(ratio = 0.618) {
    this.ratio = ratio;
  }
  calculate({value, min}) {
    const result = [];
    let remainingValue = value;
    while (remainingValue > min) {
      const nextValue = remainingValue * this.ratio;
      result.push(nextValue);
      remainingValue -= nextValue;
    }
    return result;
  }
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  GoldenRatioCalculator
});

File: packages/hquant-go-v2/common/golden_ratio_calculator.go

package common

// GoldenRatioCalculator calculates values based on the golden ratio.
type GoldenRatioCalculator struct {
	ratio float64
}

// NewGoldenRatioCalculator creates a new GoldenRatioCalculator.
func NewGoldenRatioCalculator(ratio float64) *GoldenRatioCalculator {
	return &GoldenRatioCalculator{
		ratio: ratio,
	}
}

// Calculate calculates a series of values based on the golden ratio.
func (grc *GoldenRatioCalculator) Calculate(value, min float64) []float64 {
	var result []float64
	remainingValue := value
	for remainingValue > min {
		nextValue := remainingValue * grc.ratio
		result = append(result, nextValue)
		remainingValue -= nextValue
	}
	return result
}
